src/store/modules/shipment/actions.ts

```diff
@@ -26,6 +33,67 @@ const actions: ActionTree<ShipmentState, RootState> = {
     }
     return resp;
   },
+  async fetchStatus({ state, commit }, statusIds) {
+    let resp;
+
+    const cachedStatus = JSON.parse(JSON.stringify(state.status));
+    const statusIdFilter = statusIds.reduce((filter: Array<string>, statusId: any) => {
+      if (!cachedStatus[statusId]) {
+        filter.push(statusId)
+      }
+      return filter;
+    }, []);
+
+    if (statusIdFilter.length <= 0) return cachedStatus;
+
+    try {
+      resp = await ShipmentService.fetchStatus({
+        "entityName": "StatusItem",
+        "noConditionFind": "Y",
+        "distinct": "Y",
+        "viewSize": 100,
+        "inputFields": {
+          "statusId": statusIdFilter,
+          "statusId_op": "in"
+        },
+        "fieldList": ["statusId", "description"],
+      })
+
+      if (resp.status === 200 && !hasError(resp) && resp.data?.count) {
+        const statuses = resp.data.docs;
+        statuses.map((status: any) => {
+          cachedStatus[status.statusId] = status.description
+        })
+        commit(types.SHIPMENT_STATUS_UPDATED, cachedStatus);
+      } else {
+        console.error(resp);
+      }
+    } catch(err) {
+      console.error('Something went wrong while fetching status for shipments')
+    }
+    return cachedStatus;
+  },
+  async fetchItemCount({commit}, shipmentId){
```

Comment:
We should implement it as method instead of action if we are not performing any operations related to state